### DISCUSSION TITLE
File format

### DISCUSSION THREAD

**Discussion by chadcrume** [alvarix/FLogger#61] [Ideas]
Regrouping on January 10.

Alvar wants to step back and reconsider file data format. Is plain text too error prone? Or too limiting on advanced features we've yet to think through?

Let's list some of these advanced features.

Issues with #feature label are relevant:
#41 
#31 
#22 
#15 

Issues with #"nice to have"

**Comment by chadcrume** [comment]
File System Access grant permission prompt is annoying.
Options?
Save data in backend
Save data in localStorage, sessionStorage, or indexedDB
Maybe Service Worker in Progressive Web App would have more persistent file system access options?

**Comment by alvarix** [comment]
### Feature List

Level 1

- publish (log or entry?) to web*
- cross device access*
- custom fields (can be a linked across logs. can be added on the log level or the entry level)
- copy last entry 
- import/export
- rich text

Level 2
- time tracker
- encrypted
- siri/alexa access


Level 3
- import garmin/strava etc
- geo-tagger

**Reply by alvarix** [reply]
@alvarix create user stories for first two L1.

**Reply by alvarix** [reply]
### publish (log or entry?) to web*
From within a log view or note view or edit interface I can click a button that generates a URL which is copied to clipboard. 
The URL is to a webpage of markdown or a dropbox link to view the plain text file.

### cross device access*
I access the program through a web page or pseudo app on my phone, computer or other device. 
Or at the least I can access and manipulate the app's portable data through plain text files.
Presumably there is authentication that can be stored per device.

@chadcrume are these specific enough?

**Comment by chadcrume** [comment]
Implement Dropbox authentication using the PKCE flow, with the permission type that sets up the app with a dedicate folder.

Load all files from that folder in list in a view.
User selects a file.
Load file and dump character content to screen.

Use plain text with custom delimiters:
```
1/22/27
This is one entry. It can multiple paragraphs
Like this
--//--
1/22/27
This is two entry
```
